mergify_engine/branch_updater.py

```diff
@@ -99,15 +98,39 @@ def update(pull, installation_id, method="merge"):
                 # So, retrying with the whole git history for now
                 git("fetch", "--quiet", "origin", head_branch)
                 git("fetch", "--quiet", "upstream", base_branch)
-                _do_update(git, method, base_branch, head_branch)
+                _do_update_branch(git, method, base_branch, head_branch)
             else:
                 raise
 
         return git("log", "-1", "--format=%H").decode().strip()
     except subprocess.CalledProcessError as e:  # pragma: no cover
-        LOG.error("update branch fail: %s", e.output, pull_request=pull,
-                  exc_info=True)
+        for message in AUTHENTICATION_FAILURE_MESSAGES:
+            if message in e.output:
+                raise AuthentificationFailure(e.output)
+        else:
+            LOG.error("update branch failed: %s", e.output, pull_request=pull,
+                      exc_info=True)
     except Exception:  # pragma: no cover
-        LOG.error("update branch fail", pull_request=pull, exc_info=True)
+        LOG.error("update branch failed", pull_request=pull, exc_info=True)
     finally:
         git.cleanup()
+
+
+def update(pull, installation_id, method="merge"):
+    redis = utils.get_redis_for_cache()
+
+    subscription = sub_utils.get_subscription(redis, installation_id)
+
+    if not subscription:
+        LOG.error("unable to update branch: subscription is missing")
```

Comment:
nit: I feel like `subscription` is misleading because it'd make me think that the problem is that`subscription` is `False` where this just means that there's a 404 or the like on the installation from the dashboard.

Reply:
In fact, this cannot occurs, `get_subscription` always returns a `dict`